system/lib/dlmalloc.c

```diff
@@ -18,12 +18,6 @@
 /* XXX Emscripten Tracing API. This defines away the code if tracing is disabled. */
 #include <emscripten/trace.h>
 
-/* Make malloc() and free() threadsafe by securing the memory allocations with pthread mutexes. */
-#if __EMSCRIPTEN_PTHREADS__
-#define USE_LOCKS 1
-#define USE_SPIN_LOCKS 0 // Ensure we use pthread_mutex_t.
-#endif
-
```

Comment:
I think I'd rather see this land as a seperate PR since it doesn't just effect WASM_WORKERS.

Its also rather involved to figure out what this change does.   As far as I can tell it means that USE_SPIN_LOCKS gets defined to 1 by default which ends up also defining USE_LOCKS, and then USE_SPIN_LOCKS ends up using `__sync_lock_test_and_set` and `__sync_lock_release` for locking instead of pthead_mutex_t?   Is that right?

Anyway it seems involved enough for its own PR I think..

Reply:
I don't think it is a productive use of time to artificially split to fix Wasm Workers to first work with emmalloc, and then with dlmalloc.

This change just moves the definition of `USE_LOCKS` and `USE_SPIN_LOCKS` from dlmalloc.c into system_libs.py. I had to do that migration for the following reason:

Earlier with system_libs.py, I had decided to save on the permutational explosion by not building each of the four combinations of {no threads, pthreads} X {no wasm workers, wasm workers} of system libraries, but instead the combo "pthreads but no wasm workers" is omitted as considered to be redundant, so we only build the three variants "singlethreaded" (no thread safety), "wasm workers" (use atomics and shmem so user can link in wasm workers), and "wasm workers+pthreads" (use atomics, shmem and pthread API, user can link in wasm workers) of each multithreaded system library.

However now with dlmalloc, it looks like we find a quirk that we do want to configure it to recognize only when "pthreads but not wasm workers" is being used. This is because that is the scenario when we want to use pthread mutex locks in dlmalloc. When building with pthreads+wasm workers we want to use spinlocks because malloc could be called from wasm workers that won't be able to use pthread mutex api.

Maybe there will be more code in the future that requires such special case, and we do need to build all the four variants - but so far it seems this is the only place, so opted to do this configuration specifically for libmalloc only to keep embuilder build times smaller.